opm/parser/eclipse/EclipseState/Grid/Box.hpp

```diff
@@ -82,15 +83,15 @@ namespace Opm {
 
     private:
         void init(int i1, int i2, int j1, int j2, int k1, int k2);
-        void initIndexList();
+        void initIndexList() const;
         const EclipseGrid& grid;
         size_t m_stride[3];
         size_t m_dims[3] = { 0, 0, 0 };
         size_t m_offset[3];
 
         bool   m_isGlobal;
-        std::vector<cell_index> m_active_index_list;
-        std::vector<cell_index> m_global_index_list;
+        mutable std::optional<std::vector<cell_index>> m_active_index_list;
+        mutable std::optional<std::vector<cell_index>> m_global_index_list;
```

Comment:
You seem to be a big fan of std::optional. What is the advantage of using it instead of just `std::vector` and using std::vector::empty` to test whether to init or not, and `std::vector::clear`followed by `std::vector::shrink_to_fit` to release memory.

Reply:
> You seem to be a big fan of std::optional

Yes that is true - I really like it. I think `std::optional<...>` communicates the question of whether something is initialized or not -  better than special values like `vector.empty()`. In this particular case the `initIndexList()` *should* really only be called once - so the initial clear/reset is really just for future proofing.